cdci_data_analysis/flask_app/app.py

```diff
@@ -150,8 +150,30 @@ def meta_data_src():
 
 @app.route("/download_products", methods=['POST', 'GET'])
 def download_products():
-    query = InstrumentQueryBackEnd(app, download_products=True)
-    return query.download_products()
+    from_request_files_dir = request.args.get('from_request_files_dir', 'False') == 'True'
+    download_file = request.args.get('download_file', 'False') == 'True'
+    download_products = request.args.get('download_products', 'True') == 'True'
+    query = InstrumentQueryBackEnd(app, download_products=download_products, download_files=download_file)
+    return query.download_file(from_request_files_dir=from_request_files_dir)
+
+
+@app.route("/download_file", methods=['POST', 'GET'])
+def download_file():
+    if app.config['conf'].products_url is not None and validators.url(app.config['conf'].products_url):
```

Comment:
```suggestion
    if validators.url(app.config['conf'].products_url):
```

Reply:
This fails in the case `products_url` is not present in the configuration, which is a case we considered when configuring the redirection for the `instr-list` endpoint.

Do we ever have cases where products_url is not in the configuration?